descopesdk/src/main/java/com/descope/internal/routes/Passkey.kt

```diff
@@ -0,0 +1,326 @@
+                cont.resume(Pair(code, intent))
+            }
+        }
+        val registerResponse = prepareRegisterResponse(result.first, result.second)
+        client.passkeyAddFinish(startResponse.transactionId, registerResponse)
+    }
+
+    private suspend fun finishSignUp(transactionId: String, resultCode: Int, intent: Intent?): AuthenticationResponse {
+        val jsonResponse = prepareRegisterResponse(resultCode, intent)
+        val jwtResponse = client.passkeySignUpFinish(transactionId, jsonResponse)
+        return jwtResponse.convert()
+    }
+
+    private suspend fun finishSignIn(transactionId: String, resultCode: Int, intent: Intent?): AuthenticationResponse {
+        val jsonResponse = prepareAssertionResponse(resultCode, intent)
+        val jwtResponse = client.passkeySignInFinish(transactionId, jsonResponse)
+        return jwtResponse.convert()
+    }
+}
+
+// Fido2 Helpers
+
+const val RESULT_CANCELED = 0
+
+private suspend fun performRegister(context: Context, options: String): PendingIntent {
+    val client = Fido.getFido2ApiClient(context)
+    val opts = parsePublicKeyCredentialCreationOptions(convertOptions(options))
+    val task = client.getRegisterPendingIntent(opts)
+    return task.await()
+}
+
+private suspend fun performAssertion(context: Context, options: String): PendingIntent {
+    val client = Fido.getFido2ApiClient(context)
+    val opts = parsePublicKeyCredentialRequestOptions(convertOptions(options))
+    val task = client.getSignPendingIntent(opts)
+    return task.await()
+}
+
+private fun prepareRegisterResponse(resultCode: Int, intent: Intent?): String {
+    val credential = extractCredential(resultCode, intent)
+    val rawId = credential.rawId.toBase64()
+    val response = credential.response as AuthenticatorAttestationResponse
+    return JSONObject().apply {
+        put("id", rawId)
+        put("type", PublicKeyCredentialType.PUBLIC_KEY.toString())
+        put("rawId", rawId)
+        put("response", JSONObject().apply {
+            put("clientDataJson", response.clientDataJSON.toBase64())
+            put("attestationObject", response.attestationObject.toBase64())
+        })
+    }.toString()
+}
+
+private fun prepareAssertionResponse(resultCode: Int, intent: Intent?): String {
+    val credential = extractCredential(resultCode, intent)
+    val rawId = credential.rawId.toBase64()
+    val response = credential.response as AuthenticatorAssertionResponse
+    return JSONObject().apply {
+        put("id", rawId)
+        put("type", PublicKeyCredentialType.PUBLIC_KEY.toString())
+        put("rawId", rawId)
+        put("response", JSONObject().apply {
+            put("clientDataJson", response.clientDataJSON.toBase64())
+            put("authenticatorData", response.authenticatorData.toBase64())
+            put("signature", response.signature.toBase64())
+            put("userHandle", response.userHandle?.toBase64() ?: "")
```

Comment:
I think it must be omitted if the value isn't provided
```suggestion
            response.userHandle?.let { put("userHandle", it.toBase64()) }
```